src/lib/journey/_utilities/metadata.utilities.ts

```diff
@@ -100,5 +101,11 @@ export function buildStepMetadata<T = unknown>(
         ...stageMetadata,
       },
     }),
+    // stageName and stateMetadata are mutually exclusive
+    ...(stageName && {
+      platform: {
+        stageName,
+      }
+    }),
```

Comment:
Adding `stageName`, if the stage attribute is used in the classic way, to the step metadata object. This is separate from the stage attribute used in the "new" way with JSON.